source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -86,3 +86,87 @@ as shown in the following code:
 
       DataStorageRecord[productName=1TB SSD, capacity=1.71]
 
+
+.. _fundamentals-records-annotations:
+
+Specify Field Conversion Using Annotations
+------------------------------------------
+
+This section describes the annotations you can use to configure the
+serialization behavior of record fields and provides an example to demonstrate
+the annotation behavior.
+
+.. tip::
+
+   The ``org.bson.codecs.records.annotations`` package is deprecated. Use the
+   equivalent ones from the `org.bson.codecs.pojo.annotations <{+api+}/apidocs/bson/org/bson/codecs/pojo/annotations/package-summary.html>`__
+   package instead.
+
+You can use the following annotations on record fields:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 10
+
+   * - Annotation Name
+     - Description
+
+   * - ``BsonId``
+     - Specifies the field to serialize as the _id property.
+
+   * - ``BsonProperty``
+     - Specifies a custom document field name when converting the record
+       field to BSON. Accepts the field name as the parameter.
+
+   * - ``BsonRepresentation``
+     - Specify a BSON type to store when different from the record field type.
+       Accepts the BSON type as the parameter.
+
+Example Annotated Record
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The code examples in this section reference the following sample record, which
+describes a network device:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/records/NetworkDeviceRecord.java
+   :language: java
+   :start-after: start networkDeviceRecord
+   :end-before: end networkDeviceRecord
+
+Insert an Annotated Record
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can insert a ``DataStorageRecord`` instance as shown in the following code:
+
+.. code-block:: java
+
+   MongoCollection<NetworkDeviceRecord> collection = database.getCollection("network_devices", NetworkDeviceRecord.class);
+
+   // insert the record
+   collection.insertOne(new NetworkDeviceRecord("fedcba09876543210abcdef0", "Enterprise Wi-fi", "router"));
```

Comment:
issue/question: We typically recommend that users [let the driver automatically create ObjectID values](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/crud/write-operations/insert/#a-note-about-_id) unless they have their own system with strong guarantees for uniqueness. 

Is it necessary here to provide value? If so, should include a note about this?

Reply:
I don't think it's necessary; while I think a case could be made for either including or omitting it, I think it might be handled in the description of [the related PR](https://github.com/mongodb/docs-java/pull/280) that I should have duplicated in this PR description:

> Given the similarities of the Records Page and the [POJOs page](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/data-formats/document-data-format-pojo/), I plan to ticket future work to determine the best way to organize the information to reduce duplication. This page doesn't contain as much detail as the POJOs page, but directs to other pages that should help the reader locate the omitted info.

The reason for providing the value for the id is to demonstrate how the BsonRepresentation annotation works. Without using a custom Codec, I don't think there are many other options in which specifying a type does not seem redundant. Designing and documenting a custom Codec would be duplicative of the POJO page.

This does remind me that I forgot to include the extended JSON representation of the document once inserted to show how the BsonProperty changed the field name.

Reply:
Makes sense to me.